src/Microsoft.ML.Core/Data/Schema.cs

```diff
@@ -99,53 +115,106 @@ public sealed class Column
             /// </summary>
             public Metadata Metadata { get; }
 
-            public Column(string name, ColumnType type, Metadata metadata)
+            internal Column(string name, int index, bool isHidden, ColumnType type, Metadata metadata)
             {
-                Contracts.CheckNonEmpty(name, nameof(name));
-                Contracts.CheckValue(type, nameof(type));
-                Contracts.CheckValueOrNull(metadata);
+                Contracts.AssertNonEmpty(name);
+                Contracts.Assert(index >= 0);
+                Contracts.AssertValue(type);
+                Contracts.AssertValueOrNull(metadata);
 
                 Name = name;
+                Index = index;
+                IsHidden = isHidden;
                 Type = type;
-                Metadata = metadata;
+                Metadata = metadata ?? Metadata.Empty;
             }
 
             public override string ToString()
             {
-                var metadataString = (Metadata == null || Metadata.Schema.ColumnCount == 0) ?
-                    null : $" {{{string.Join(", ", Metadata.Schema.GetColumns().Select(x => x.column.Name))}}}";
+                var metadataString = (Metadata == null || Metadata.Schema.Count == 0) ?
+                    null : $" {{{string.Join(", ", Metadata.Schema.Select(x => x.Name))}}}";
                 return $"{Name}: {Type}{metadataString}";
             }
         }
 
+        /// <summary>
+        /// This class represents the schema of one column of a data view, without an attachment to a particular <see cref="Schema"/>.
+        /// </summary>
+        public sealed class DetachedColumn
```

Comment:
>class [](start = 22, length = 5)

Did we not discuss that `Column` should be a struct? And if so, it seems like all the more reason why `DetachedColumn` should be a struct, since the arguments in favor of that seemed to apply?